Assets/Scripts/Weapon/HarpoonGun/HarpoonSpear.cs

```diff
@@ -0,0 +1,127 @@
+using Unity.VisualScripting;
+using UnityEngine;
+using Enemies;
+using Omnia.Utils;
+/*
+    The projectile for HarpoonGun
+    This class should only be interacted upon by its prefabs and HarpoonGun
+*/
+public class HarpoonSpear : MonoBehaviour {
+    public LayerMask enemyLayer;
+    public LayerMask playerLayer;
+    public LayerMask groundLayer;
+
+    public Collider2D Collider2D;
+    public Rigidbody2D Rigidbody2D;
+
+    private bool dropped;
+    private HarpoonGun gun;
+
+    // Tracking enemy
+    private Enemy taggedEnemy;
+    public Enemy TaggedEnemy {
+        get { return taggedEnemy; }
+    }
+
+    public void Awake() {
+        this.dropped = false;
+        this.taggedEnemy = null;
+    }
+
+    public void OnEnable() {
+        Enemy.Death += HandleEnemyDeath;
+    }
+
+    public void OnDisable() {
+        Enemy.Death -= HandleEnemyDeath;
+    }
+
+    // Fires the spear in the rotation of the gun with its velocity
+    public void Fire(HarpoonGun gun) {
+        this.dropped = false;
+        this.gun = gun;
+
+        Rigidbody2D.velocity = gun.transform.right * gun.harpoonVelocity;
```

Comment:
Wait, does right work when the gun is flipped?

Reply:
I think so. If you think about the angle of the gun as your mouse rotates counter-clockwise, when the player turns from right to left the sprite of the gun actually flips on the Y axis, so "right" always points towards the end of the gun.